pkg/utils/download.go

```diff
@@ -12,24 +12,27 @@
 package utils
 
 import (
+	"context"
 	"net/url"
 	"os"
 	"path"
 	"strings"
 	"time"
+
+	"github.com/google/go-github/github"
 )
 
 // DownloadFromURLThenExtract downloads files from a URL
 // to a destination, extracting them if necessary
-func DownloadFromURLThenExtract(URL string, destination string) error {
+func DownloadFromURLThenExtract(URL, destination string, gitCredentials GitCredentials) error {
 	if _, err := url.ParseRequestURI(URL); err != nil {
 		return err
 	}
 
 	if IsTarGzURL(URL) {
 		return DownloadFromTarGzURL(URL, destination)
```

Comment:
Do we need to supply credentials when downloading a tar.gz? (I can't remember which project type uses this instead of zip, it may be Appsody.)

Reply:
I didn't realise - should I make that change in this or the next PR?